ark/phenotyping/cell_cluster_utils.py

```diff
@@ -477,17 +475,23 @@ def cluster_cells(base_dir, cell_pysom, cell_som_cluster_cols,
     if cell_pysom.weights is None:
         raise ValueError("Using untrained cell_pysom object, please invoke train_cell_som first")
 
+    # non-pixel cluster inputs won't be cell size normalized
+    cols_to_drop = ['fov', 'segmentation_label']
+    if 'cell_size' in cell_pysom.cell_data.columns.values:
+        cols_to_drop.append('cell_size')
+
     # ensure the weights columns are valid indexes, do so by ensuring
     # the cluster_counts_norm and weights columns are the same
     # minus the metadata columns that appear in cluster_counts_norm
     cluster_counts_size_norm = cell_pysom.cell_data.drop(
-        columns=['fov', 'segmentation_label', 'cell_size']
+        columns=cols_to_drop
     )
 
-    misc_utils.verify_same_elements(
-        enforce_order=True,
-        cluster_counts_size_norm_columns=cluster_counts_size_norm.columns.values,
-        cell_weights_columns=cell_pysom.weights.columns.values
+    # handles the case if user specifies a subset of columns for generic cell clustering
+    # NOTE: CellSOMCluster ensures column ordering by using the preset self.columns as an index
+    misc_utils.verify_in_list(
+        cell_weights_columns=cell_pysom.weights.columns.values,
+        cluster_counts_size_norm_columns=cluster_counts_size_norm.columns.values
```

Comment:
For lines 506-onwards, since this isn't necessary for the generic clustering, we should pull this out into its own function or add a flag

Reply:
Agreed we can pull this out into its own function. However, we still need to compute average number of each cell SOM cluster per cell since these are the inputs for ConsensusClustering.

Reply:
Can you explain what the inputs to consensus clustering are in the current notebook 3, not this version? What's the shape, columns, etc?

Reply:
The inputs to consensus clustering is a table in the following format:

| cell_som_cluster | pixel_meta_cluster_1 | pixel_meta_cluster_2 | ... | pixle_meta_cluster_n | count |
| --| -- | -- | -- | -- | -- |
| 1  | ... | ... | ... | ... | ... |
| 2 | ... | ... | ... | ... | ... | 
| 3 | ... | ... | ... | ... | ... |
| ... | ... | ... | ... | ... | ... |
| 100 | ... | ... | ... | ... | ... |

Each entry is the average number of pixel meta clusters found per cell SOM cluster. Training is done on these average count columns, the `count` column is needed to visualize the barchart at the top of the remapping step.

The general idea is we need, for each cell SOM cluster, the average expression of the columns used for training. I'm starting to understand the concern now, if the original expression columns are prediction values instead of counts, the concept of "average predicted probability per cell SOM cluster" might not make as much sense.

Let me know what you think. Maybe @cliu72 and/or @JLrumberger can comment on this further.

Reply:
Yes exactly. The equivalent here would be the average value of each column in each cell cluster, rather than the average count of each column.

Reply:
@ngreenwald the existing functionality already computes this. We just need to change the nomenclature used in functions and variable names to avoid confusion.

Reply:
Sounds good